src/go/rpk/pkg/cli/cmd/redpanda/config.go

```diff
@@ -171,9 +174,8 @@ func bootstrap(fs afero.Fs) *cobra.Command {
 		&id,
 		"id",
 		-1,
-		"This node's ID (required).",
+		"This node's ID. If unset, Redpanda will assign one automatically.",
```

Comment:
IMO we should hide this flag now. What is the benefit to letting users set this?

Reply:
This is a good question -- I assumed it was required for compatibility, but it is something we should steer folks away from using. I'm happy to make that change, but I think it should be in a separate change, perhaps after the feature has time to bake

Reply:
Marking a flag hidden doesn't remove the capability, it removes it from being displayed in the help text. Existing uses of --id will continue to work

Reply:
Done